### PR TITLE
README: Add Assumptions, Installation, and Command Line Usage notes

### PR DESCRIPTION
**Pull request by christophermanning**
cc @billwanjohi

### CODE REVIEW

**Comment by captainkerk** [comment]
LGTM.

File: README.md [hunk 1]
```diff
@@ -6,11 +6,30 @@ to allow codebases already using AWS credentials to use IAM roles with minimal t
 refactoring. In the same vein, the Role Injector can also be used to help users using the
 command line tools to assume a role.
 
+## Assumptions
+
+ - At least two AWS Accounts:
+   - AWS Account 1 must have a policy that includes sts:AssumeRole to AWS Account 2
+   - AWS Account 2 must have a Trust Relationship on a role that references AWS Account 1
```
Comment (billwanjohi) [inline README.md:18]: This is not strictly necessary. Even if you have one AWS account, you could use this to maintain very restricted IAM permissions for non-MFA authenticated requests, and then assume a role in that same account to do more sensitive work.

**Comment by christophermanning** [comment]
LGTM

File: README.md [hunk 2]
```diff
@@ -6,11 +6,42 @@ to allow codebases already using AWS credentials to use IAM roles with minimal t
...
+ - AWS CLI configured correctly, storing 'aws_access_key_id' and
+   'aws_secret_access_key' in either environment variables -OR- in
+   ~/.aws/credentials
+ - One of the following Scenarios apply:
+
+### Scenario One: Federated AWS Accounts
+ - At least two AWS Accounts:
+   - AWS Account 1 must have a policy that includes sts:AssumeRole to AWS Account 2
+   - AWS Account 2 must have a Trust Relationship on a role that references AWS Account 1
+ - AWS Account 1 may now assume the the role in AWS Account 2 that has the Trust Relationship
```
Comment (billwanjohi) [inline README.md]: > the the

File: README.md [hunk 3]
```diff
@@ -6,11 +6,42 @@ to allow codebases already using AWS credentials to use IAM roles with minimal t
...
+
+### Scenario Two: Single AWS Account:
+ - An IAM User Account with a policy that include sts:Assume on a IAM
```
Comment (billwanjohi) [inline README.md]: *includes sts:Assume on *an

File: README.md [hunk 4]
```diff
@@ -6,11 +6,42 @@ to allow codebases already using AWS credentials to use IAM roles with minimal t
...
+   Role.
+ - The IAM Role has a policy that allows the IAM User to assume it
```
Comment (billwanjohi) [inline README.md]: s/policy that allows the IAM User/trust relationship that allows entities in the account

**Pull request merged**